integrations/oci/oci-secrets-config-source/src/main/java/io/helidon/integrations/oci/secrets/configsource/OciSecretsConfigSourceProvider.java

```diff
@@ -0,0 +1,526 @@
+                                return null;
+                            });
+                        java.util.Date d = ss.getTimeOfCurrentVersionExpiry();
+                        Instant i = d == null ? null : d.toInstant();
+                        if (i != null && (mostDistantExpirationInstant == null || mostDistantExpirationInstant.isBefore(i))) {
+                            mostDistantExpirationInstant = i;
+                        }
+                    }
+                    SecretBundleConfigSource.this.mostDistantExpirationInstant = mostDistantExpirationInstant; // volatile write
+                    completeAllSecretsTasks(tasks, secrets);
+                    ObjectNode.Builder onb = ObjectNode.builder();
+                    for (Entry<String, ValueNode> e : valueNodes.entrySet()) {
+                        onb.addValue(e.getKey(), e.getValue());
+                    }
+                    return Optional.of(NodeContent.builder()
+                                       .node(onb.build())
+                                       .build());
+                };
+            }
+        }
+
+
+        /*
+         * Instance methods.
+         */
+
+
+        @Deprecated // For use by the Helidon Config subsystem only.
+        @Override // PollableSource
+        public boolean isModified(Instant instant) {
+            Instant i = this.mostDistantExpirationInstant;
+            return i == null || i.isBefore(instant == null ? now() : instant);
+        }
+
+        @Deprecated // For use by the Helidon Config subsystem only.
+        @Override // NodeConfigSource
+        public Optional<NodeContent> load() {
+            return this.loader.get();
+        }
+
+        @Deprecated // For use by the Helidon Config subsystem only.
+        @Override // PollableSource
+        public Optional<PollingStrategy> pollingStrategy() {
+            return super.pollingStrategy();
+        }
+
+        private Optional<NodeContent> absentNodeContent() {
+            this.mostDistantExpirationInstant = null; // volatile write
+            return ABSENT_NODE_CONTENT;
+        }
+
+
+        /*
+         * Static methods.
+         */
+
+
+        private static Builder builder() {
+            return new Builder();
+        }
+
+        private static void completeAllSecretsTasks(Collection<? extends Callable<Void>> tasks, AutoCloseable secrets) {
+            RuntimeException re = null;
+            try (ExecutorService es = newVirtualThreadPerTaskExecutor()) {
+                for (Future<?> future : es.invokeAll(tasks)) {
+                    try {
+                        future.get();
+                    } catch (RuntimeException e) {
+                        if (re == null) {
+                            re = e;
+                        } else {
+                            re.addSuppressed(e);
+                        }
+                    } catch (Exception e) {
```

Comment:
please replace with `catch (InterrupteedExceptione)`
the block cannot throw any other checked exception, and you catch all runtime exceptions in the previous block

Reply:
Good catch; I missed that `ExecutorService` overrides `close()` to not throw `Exception`.

Reply:
Done